migrateBuckets/migrateBuckets.js

```diff
@@ -0,0 +1,327 @@
+                    return next(err);
+                }
+                const newObjectName = _createUsersBucketKey(
+                    bucketName, destCanonicalId);
+                return next(null, newObjectName, objectMD);
+            }),
+            (newObjectName, objectMD, next) => mdclient.putObject(
+            metaBucket, newObjectName, objectMD, serUids, err => {
+                if (err) {
+                    log.error('Error updating users..bucket object ' +
+                        `${matchingBucket.key}: ${err}`);
+                    return next(err);
+                }
+                log.debug(`users..bucket object ${matchingBucket.key} ` +
+                    'successfully updated');
+                return next();
+            }),
+            next => mdclient.deleteObject(metaBucket, matchingBucket.key,
+            serUids, err => {
+                if (err) {
+                    log.error('Error deleting old bucket entry from ' +
+                        `users bucket: ${err}`);
+                } else {
+                    log.debug('Successfully deleted old bucket entry ' +
+                        'from users bucket');
+                }
+                return next();
+            }),
+        ], err => {
+            if (err) {
+                return cb(err);
+            }
+            log.debug('Successfully updated users..bucket entry');
+            return cb();
+        });
+    });
+}
+
+function _changeOneObjectAccount(bucket, object, log, cb) {
+    const {
+        bucketName,
+        destCanonicalId,
+        destOwnerName,
+    } = bucket;
+    const { Key, VersionId } = object;
+    const decodedId = VersionId ? versionIdUtils.decode(VersionId) : undefined;
+
+    // if versiondId is undefined, retrieval will return most current object md
+    const getOptions = { versionId: decodedId };
+
+    mdclient.getObject(bucketName, Key, log.getSerializedUids(),
+    (err, data) => {
+        const objectMD = JSON.parse(data);
+        if (err) {
+            log.error(`Error getting object metadata: ${err}`);
+            return cb(err);
+        }
+        const objectId = objectMD['owner-id'];
+        const objectOwner = objectMD['owner-display-name'];
+        if (objectId === destCanonicalId && objectOwner === destOwnerName) {
+            log.debug(`Object ${Key} id already updated`);
+            return cb();
+        }
+        objectMD['owner-id'] = destCanonicalId;
+        objectMD['owner-display-name'] = destOwnerName;
+        const putOptions = {};
+        if (VersionId) {
+            putOptions.versionId = VersionId;
+        }
+        return mdclient.putObject(bucketName, Key, JSON.stringify(objectMD),
+        log.getSerializedUids(), err => {
+            if (err) {
+                log.error(`Error updating object ${Key} metadata: ${err}`);
+                return cb(err);
+            }
+            log.debug(`Object ${Key} id successfully updated`);
```

Comment:
I'd suggest to make it a `log.info` call: it can be a lot of log, depending on the bucket size, but it's better than no feedback at all until the end of the migration.

Also you can add the `VersionId` field to the log line along with the `Key`